lib/bundler/source.rb

```diff
@@ -38,5 +38,17 @@ def include?(other)
     def inspect
       "#<#{self.class}:0x#{object_id} #{self}>"
     end
+
+  private
+
+    def version_color(spec_version, locked_spec_version)
+      earlier_version?(spec_version, locked_spec_version) ? :yellow : :green
+    end
+
+    def earlier_version?(spec_version, locked_spec_version)
+      Gem::Version.new(spec_version) < Gem::Version.new(locked_spec_version)
+    rescue ArgumentError
```

Comment:
@segiddins We’re turning the `version` strings in the parameters into `Gem::Version` so that we can use the `<` operator (and not have to write multi-decimal-point comparison logic from scratch).

However, some of the time when I was testing it, the strings would be dependencies rather than versions -- e.g., `>= 2.0.0` rather than `2.1.0`. In those cases, `Gem::Version.new` will throw an exception, and we simply catch that and default to not adjusting the default color.

If there’s a cleaner implementation, I’m certainly open to that. Hope this makes the existing implementation make a bit more sense in the meantime.